src/components/utils/webcam.js

```diff
@@ -0,0 +1,45 @@
+import enumerateDevices from 'enumerate-devices'
+
+const enumerateDevicesInternal = (onSuccess, onError) => {
+  try {
+    enumerateDevices().then(onSuccess).catch(onError);
+  }
+  catch (exception){
+    onError(exception)
+  }
+}
+
+const checkDevicesInfo = checkFn =>
+  onResult =>
+    enumerateDevicesInternal(
+      devices => onResult(checkFn(devices)),
+      () => onResult(false)
+    )
+
+const isVideoDevice = ({ kind = '' }) => kind.includes('video')
+
+const hasDevicePermission = ({ label }) => !!label
+
+export const checkIfHasWebcam = checkDevicesInfo(
+  devices => devices.some(isVideoDevice)
+)
+
+export const checkIfWebcamPermissionGranted = checkDevicesInfo(
+  devices => devices.filter(isVideoDevice).some(hasDevicePermission)
+)
+
+const { mediaDevices = {} } = navigator;
+const { getUserMedia = () => Promise.reject() } = mediaDevices;
+const noop = () => {}
+
+export const detectWebcamMaxSupportedHeight = heights =>
+  heights.reduce((promise, height) =>
+    promise.then(maxSupportedHeight =>
+      getUserMedia.call(mediaDevices, { video: { height }, audio: false }, noop, noop)
```

Comment:
for the browsers that always ask for permissions, we need to test carefully on whether this approach ask the user for permission multiple times or not. Because from my deduction, even if the resolution is accepted in the first attempt, the `getUserMedia` will still have two calls, no?